docs/code-review-guidelines.md

```diff
@@ -0,0 +1,42 @@
+# Code review guidelines
+
+## Background
+
+We use pull requests (PRs) for all code changes. All PRs have to be
+reviewed and approved by at least one other developer before being merged into
+`master`.
+
+The main objective of code reviews is to improve the quality of code that
+reaches production, so that our codebase is maintainable and correct. We want
+to reduce the number of bugs that go live though there are other secondary
+benefits like knowledge sharing.
+
+## Guidelines
+
+### Preparing PRs
+
+- Create a draft PR if you want early feedback for work in progress.
+- We have a template for the PR descriptions. The _Guidance to
+  review_ section should contain any information that a reviewer might find
+  helpful. In particular consider adding:
+  - A list of particular questions that you would like feedback on, e.g. design
+    choices.
+  - Step-by-step instructions on how to manually test the change, whether
+    running locally or using a review app.
+- Consider extending test application data if it would make the PR easier to test.
+- If a PR is going to be complex to review, affects core functionality, touches
+  on application security or is deemed to be risky for some other reason tag
+  it as requiring two approvals rather than the normal one.
+
```

Comment:
I was wondering if it would be useful to include guidance about commit history.
Some recent PRs on the project had a commit consisting of a single breaking test. This can make it difficult to  find when a breaking change was introduced with `git bisect`.  Also normally I don't think correcting linting  errors picked by rubocop deserves a separate commit if the errors were introduced in the same PR. 

```suggestion
- Follow general best practice for raising PR e.g. [How to raise a good pull request](https://www.annashipman.co.uk/jfdi/good-pull-requests.html).
  - Aim for each commit to be atomic, introducing a non-breaking change with all tests and linter passing. Consider arranging your commits into appropriate logical chunks, which can make the PR easier to review and revert, with [git's history rewriting features](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History).
  - Try to avoid merge commits and use `git rebase master` instead.
```